### ISSUE TITLE
ENH: Provide functions to compute log-integrals numerically (e.g., log_simpson, log_trapezoid, ..) 

### ISSUE THREAD

**Issue by adrn** [scipy/scipy#15613]
### Is your feature request related to a problem? Please describe.

In some data analysis contexts, I have needed to compute integrals over density functions that have steep density profiles, in order to compute the log of the integral (to include in, e.g., a log-likelihood function). In these cases, for numerical stability, I have often written the density functions to compute `log_density()` instead, so procedurally I have had to do something like `np.log(simpson(np.exp(ln_density(x)), x=x)`. This isn't very precise and I have run into roundoff issues when integrating over large domains.

### Describe the solution you'd like.

Since Simpson's rule and many of the other numerical integration methods that work on grids essentially do a sum of elements of the input array (weighted by coefficients), it occurred to me that these parts could be replaced by `logsumexp()`'s and the integration methods could instead produce the log-integral given the log-values of a function. As a trial implementation, here's an example of a `log_simpson()` function: https://github.com/adrn/solar-velocity/blob/main/notebooks/integrate_helpers.py#L47

Is this something that it would be worth thinking about implementing in scipy?

**Comment by mdhaber** [comment]
Thanks for your patience. Yes, I think it sounds valuable and I think I've wanted that sort of thing before. Seems like this could help integrate in deep tails of distributions. Would it make sense to make it an argument of existing quadrature functions?

**Comment by mdhaber** [comment]
I just ran across this again. I added the feature to the new [qmc_quad](https://scipy.github.io/devdocs/reference/generated/scipy.integrate.qmc_quad.html#scipy.integrate.qmc_quad) in gh-16923. Since there hasn't been much discussion here, I don't think the issue needs to remain open, but I think a `log` parameter would be a useful addition to any of the other quadrature routines that are implemented in Python.
